File: python/1410_html-entity-parser.py
class Solution:
    def entityParser(self, text: str) -> str:

        codex = {
            "&quot;": '"',
            "&apos;": "'",
            "&amp;": "&",
            "&gt;": ">",
            "&lt;": "<",
            "&frasl;": "/"
        }

        i = 0
        while i < len(text):

            if text[i] == "&":
                for x in codex:
                    if text[i:i+len(x)] == x:
                        text = text[:i] + codex[x] + text[i+len(x):]
                        break

            i += 1

        return text

# Time:  O(n)
# Space: O(1)

File: js/1410_html-entity-parser.js
/**
 * @param {string} text
 * @return {string}
 */
 var entityParser = function(text) {
    return text
        .replace(/&gt;/g, `>`)
        .replace(/&lt;/g, `<`)
        .replace(/&frasl;/g, `/`)
        .replace(/&quot;/g, `"`)
        .replace(/&apos;/g, `'`)
        .replace(/&amp;/g, `&`)
};
